### DISCUSSION TITLE
My commits author is changed automatically to root in linux

### DISCUSSION THREAD

**Discussion by arsekil** [community/community#91777] [New to GitHub] [answered]
Hello everyone,

I am new to linux as well, and Git as well, so I don't understand why git changes the name of the author of the commits I am making to `root`, when i already set the --global user.name.

**Comment by NipulM** [comment]
Git usually uses a hierarchy of configurations. Check if there is a local configuration that overrides your global settings

```
cd /path/to/your/repo
git config user.name
git config user.email
```

If there's a local config, you can change them by doing:
```
git config user.name "user_name"
git config user.email "your_email"
```

If you are using SSH for Git operations, make sure that your SSH key is correctly associated with your Git account. If the SSH key is not recognized, Git may default to the "root" user. 👍🏼

**Reply by ragmer09toske** [reply, marked as answer]
Hello! 👋

If Git is showing your commits as authored by "root," it might be due to permissions or the way you're initializing your Git repository. 

Make sure you have the correct permissions and are in the right directory when making commits. Also, verify your local repository's config with : `git config --local user.name` to ensure it's set correctly for the specific project.

**Reply by arsekil** [reply]
Setting the local config solved the issue, i hope it persists.
